src/App.js

```diff
@@ -1,14 +1,41 @@
 import "./App.css";
 import React from "react";
 import { Router } from "./frontend/router/Router";
-import { Footer, Navbar } from "./frontend/components";
+import { Navbar, SidebarMenu } from "./frontend/components";
+import { useLocation } from "react-router-dom";
+
+const sidebarDisplay = (pathname) => {
+  if (
+    pathname === "/explore" ||
+    pathname === "/bookmarks" ||
+    pathname === "/profile" ||
+    pathname === "/feed"
+  ) {
+    return false;
+  } else {
+    return true;
+  }
+};
+
+const flexToggle = (pathname) => {
+  if (!sidebarDisplay(pathname)) {
+    return "flex--row division__container";
+  } else {
+    return "";
+  }
+};
 
 function App() {
+  const location = useLocation();
   return (
     <div className="App">
       <Navbar />
-      <Router />
-      <Footer />
+      <div className={flexToggle(location.pathname)}>
+        {!sidebarDisplay(location.pathname) && <SidebarMenu />}
```

Comment:
If possible, transfer all this conditional rendering logic to the `Sidebar` component. In this way keep the `App.js` simple to read.